server/src/main/java/org/opensearch/rest/action/admin/cluster/RestNodesStatsAction.java

```diff
@@ -175,6 +176,25 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
 
                     nodesStatsRequest.indices(flags);
                 }
+            } else if (metrics.contains("caches")) {
+                // Extract the list of caches we want to get stats for from the submetrics (which we get from index_metric)
+                Set<String> cacheMetrics = Strings.tokenizeByCommaToSet(request.param("index_metric", "_all"));
+                CommonStatsFlags cacheFlags = new CommonStatsFlags();
+                cacheFlags.clear();
+                if (cacheMetrics.size() == 1 && cacheMetrics.contains("_all")) {
+                    cacheFlags.includeAllCacheTypes();
+                } else {
```

Comment:
Do we throw exception if _all is included with specific representations?

Reply:
We don't. Can probably remove cacheMetrics.size() == 1 from this check, assuming if the user sent _all they should get it regardless of what else they sent.